### PR TITLE
date-picker

### PR DESCRIPTION
**Pull request by dgsmith2**

### CODE REVIEW

**Comment by ebrehault** [comment]
Hi @dgsmith2 ! Glad to see you're preparing a PR on Pastanaga :)

Apparently you've removed your email address from your GitHub profile so it is not associated anymore to your Contributor Agreement, and it will prevent the PR to be merged.
If that was a professional email you don't have anymore, I guess you can contact @ableeb, he should be able to reference a different email.

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss [hunk 1]
```diff
@@ -0,0 +1,99 @@
+:host {
+    --day-width: 32px;
+    --header-color: #6c707f;
+
+    .calendar-popup {
+        font-size: 14px;
+        line-height: 24px;
+        color: #001530;
```
Comment (barcafa) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss:8]: `$color-dark-stronger`?

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            }
+        }
+    }
+
+    .calendar-row ::ng-deep .pa-button {
+        width: var(--day-width);
+
+        &::ng-deep.other-month,
+        &::ng-deep.other-month:hover {
+            color: #e3e7eb;
```
Comment (barcafa) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss:61]: `$color-neutral-light`?

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        }
+    }
+
+    .calendar-list-item {
+        padding: 12px;
+        font-size: 16px;
+        cursor: pointer;
+
+        &.selected {
+            background-color: #e5f3ff;
```
Comment (barcafa) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss:86]: `$color-primary-lightest`

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        }
+
+        &:hover,
+        &.selected:hover {
+            background-color: #f2f5f7;
+        }
+    }
+
+    .calendar-row ::ng-deep pa-button.selected {
+        background-color: #e5f3ff;
```
Comment (barcafa) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss:96]: `$color-primary-lightest`

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        font-size: 16px;
+        cursor: pointer;
+
+        &.selected {
+            background-color: #e5f3ff;
+        }
+
+        &:hover,
+        &.selected:hover {
+            background-color: #f2f5f7;
```
Comment (barcafa) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss:91]: `$color-neutral-lighter`

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss [hunk 6]
```diff
@@ -0,0 +1,107 @@
+@import '../../styles/variables';
+
+:host {
+    --day-width: 32px;
```
Comment (mpellerin42) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss:4]: replace all pixel values by rhythm.

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        .spacer {
+            flex-grow: 1;
+        }
+    }
+
+    .calendar-weekdays,
+    .calendar-row {
+        display: flex;
+        color: var(--header-color);
+        padding: 6px 0;
```
Comment (mpellerin42) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss:47]: 6px is not part of our rhythm map, so there is a big chance that this value is wrong, or meant to be coming from some alignment flex rule more than a static padding.

**Review by mpellerin42 — CHANGES_REQUESTED** [review]

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.ts [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+    constructor(private cd: ChangeDetectorRef) {
+        this.formControl = new FormControl();
+        this.trackedDate = new Date();
+    }
+
+    ngOnInit(): void {
+        // honor text entry
+        this.formControl.valueChanges
+            .pipe(
+                debounceTime(500),
```
[the remark this answers is not in the document]
Comment (dgsmith2) [reply projects/pastanaga-angular/src/lib/date-picker/date-picker.component.ts:104]: I'll make this change but not 100% on board.  A constant (named) for css/animation is being used in a context that has nothing to do with it.

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.ts [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+    registerOnChange(onChange: any): void {
+        this._onChange = onChange;
+    }
+
+    registerOnTouched(onTouched: any): void {
+        this._onTouched = onTouched;
+    }
+
+    setDisabledState(disabled: boolean) {
+        this.disabled = disabled;
```
Comment (julienCarret) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.ts:292]: should we add this.formControl.disable() ?

File: projects/pastanaga-angular/src/lib/translate/i18n/en.ts [hunk 10]
```diff
@@ -1,6 +1,15 @@
...
 export const I18N_EN: TranslationEntries = {
+    'calendar': {
+        'monday': 'M',
+        'tuesday': 'T',
+        'wednesday': 'W',
+        'thursday': 'T',
+        'friday': 'F',
+        'saturday': 'S',
+        'sunday': 'S',
+    },
```
Comment (julienCarret) [inline projects/pastanaga-angular/src/lib/translate/i18n/en.ts:12]: We could have the months here to...

**Review by julienCarret — APPROVED** [review]

File: projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss [hunk 11]
```diff
@@ -74,20 +74,20 @@
         display: flex;
         align-items: center;
         color: var(--header-color);
-        font-size: 12px;
+        font-size: font-size(xs);
         font-weight: 700;
-        line-height: 16px;
+        line-height: line-height(s);
         text-transform: uppercase;
```
Comment (mpellerin42) [inline projects/pastanaga-angular/src/lib/date-picker/date-picker.component.scss:80]: seems like you could have used pastanaga typography class `.overline` here.
You can see all the typography classes in https://github.com/plone/pastanaga-angular/blob/a09a31d30635fc381ae7aeb0537052fef7aa92e6/projects/pastanaga-angular/src/styles/_typography.scss

**Review by mpellerin42 — APPROVED** [review]

**Pull request merged**